R/plots-qc.R

```diff
@@ -64,6 +64,78 @@ qc_sample_hist <- function(gimap_dataset, wide_ar = 0.75) {
   return(sample_cpm_histogram)
 }
 
+#' Create a histogram for the variance within replicates for each pgRNA
+#' @description This function uses pivot_longer to rearrange the data for plotting, finds the variance for each pgRNA construct (using row number as a proxy) and then plots a histogram of these variances
+#' @param gimap_dataset The special gimap_dataset from the `setup_data` function which contains the transformed data
+#' @param wide_ar aspect ratio, default is 0.75
+#' @importFrom tidyr pivot_longer
+#' @importFrom magrittr %>%
+#' @import ggplot2
+#' @return a ggplot histogram
+#' @examples \dontrun{
+#'
+#' }
+#'
+
+qc_variance_hist <- function(gimap_dataset, wide_ar = 0.75){
+  
+  return(
+    gimap_dataset$transformed_data$log2_cpm[,3:5] %>%
```

Comment:
What's the reasoning for picking columns 3 - 5 here? Is this going to be standard across datasets?

Reply:
I don't know if it's going to be standard across datasets :( And it's one of the conversations we've had with the Berger lab, and I'm still unsure. But when I'm working with columns 3-5, I'm assuming I'm working with last day replicates. When I'm working with the first column, I'm assuming I'm working with plasmid/day 0. Would love to have a more robust way to do that! I think requiring sample metadata is probably the best way, but even then, the sample metadata would have to be formatted in ways we expect

Reply:
At this point in the workflow, they should have metadata involved so we can make an argument that basically tells which samples to apply this filter by (so supply a vector that corresponds to the samples we want to target for this, so in this case 3-5). 

How does this idea sound to you?

Reply:
Alternatively at this point in the workflow, they should have metadata involved so we could also have them supply a column that says which columns should be used to determine a filter. But  I think doing an easier method like what I've described above is fine to start if you are cool with that.

Reply:
Love that idea! So I should just add arguments with defaults specifying the column and they can change the columns?

Reply:
Yeah so like `filter_target_col = NULL` and then
```
is.null(filter_target_col) filter_target_col <- 1:ncol(gimap_dataset$transformed_data$log2_cpm)
```

But otherwise in the code: 

```
gimap_dataset$transformed_data$log2_cpm[,filter_target_col]
```

Open to better argument name but this is the gist of what I'm thinking.

Reply:
Works for me, and I'm working on adding several so each type of filter has its own. It'll be in the next stacked PR since it's a bit of a heavier change across several files. Thank you so much!